plugins/services/src/js/components/modals/ServiceRestartModal.js

```diff
@@ -79,38 +80,64 @@ class ServiceRestartModal extends React.Component {
     );
   }
 
-  render() {
-    const { isPending, onClose, open, service, restartService } = this.props;
+  getServiceLabel() {
+    const { service } = this.props;
+
+    if (service instanceof Pod) {
+      return "Pod";
+    }
+
+    if (service instanceof ServiceTree) {
+      return "Group";
+    }
+
+    return "Service";
+  }
+
+  getModalHeading() {
+    const serviceLabel = this.getServiceLabel();
 
-    const heading = (
+    return (
       <ModalHeading>
-        Restart Service
+        Restart {serviceLabel}
       </ModalHeading>
     );
+  }
+
+  render() {
+    const { isPending, onClose, open, service, restartService } = this.props;
     let serviceName = "";
+    const serviceLabel = this.getServiceLabel();
 
     if (service) {
```

Comment:
Since you've cleaned this up a lot already - maybe we could remove this `if` statement too since service is a required prop!